test/native/integration-test-helpers/add-block.js

```diff
@@ -44,7 +44,19 @@ export const addBlock = async (
 		},
 	} );
 
-	fireEvent.press( await waitFor( () => getByText( blockName ) ) );
+	const blockButton = await waitFor( () => getByText( blockName ) );
+	// Blocks can perform belated state updates after they are inserted.
+	// To avoid potential `act` warnings, we ensure that all timers and queued
+	// microtasks are executed.
+	await withFakeTimers( async () => {
+		fireEvent.press( blockButton );
+		// Run all timers, in case any performs a state updates.
+		// Column block example: https://t.ly/NjTs
+		act( () => jest.runOnlyPendingTimers() );
+		// Let potential queued microtasks (like Promises) to be executed.
+		// Inner blocks example: https://t.ly/b95nA
+		await act( async () => {} );
+	} );
```

Comment:
I noticed that with RN `0.71.8`, there was a dramatic increase of `act` warnings when running tests ([reference](https://github.com/WordPress/gutenberg/actions/runs/5214263615/jobs/9410316091#step:5:491)). I investigated further and found out that most of them were related to block insertion, especially for blocks with inner blocks. I identified belated state updates produced by microtasks ([reference](https://github.com/WordPress/gutenberg/blob/05b94d1ff57027d53e4fed98a42cf80ca65f42aa/packages/block-editor/src/components/inner-blocks/use-nested-settings-update.js#L132-L143)). With this approach, we'll ensure that all state updates potentially produced during block creation are handled 🤞 .

@dcalhoun as we worked closely on [a similar issue with `act` warnings](https://github.com/WordPress/gutenberg/pull/38052). I'd appreciate your input in this workaround. Thanks 🙇